ERCS/erc-7565.md

```diff
@@ -0,0 +1,327 @@
+
+- Implementation Suggestion: MAY be implemented as `external` `view`.
+- Usage: Enables querying the current repay amount of a specific NFT.
+  
+This is what I write, but I think what you give me really reflect well what I want to propose. Please update what I write
+
+## Rationale
+
+### Design Motivation
+
+The design of this standard is driven by the need to address specific challenges in the DeFi sector, particularly concerning the liquidity and management of assets locked as collateral. Traditional mechanisms in DeFi often require asset holders to lock up their assets for participation in activities such as lending, staking, or yield farming, which results in a loss of liquidity. This standard aims to introduce a more flexible approach, allowing asset holders to retain some liquidity while their assets are locked, thereby enhancing the utility and appeal of DeFi products.
+
+### Design Decision
+
+- Dual-Role System (Asset Owner and DeFi Platform/Contract): A clear division is established between the NFT owner (asset holder) and the DeFi platform or contract utilizing the NFT as collateral. This distinction simplifies the management of rights and responsibilities, enhancing clarity and reducing potential conflicts.
+
+- Enhancing Liquidity without Compromising Asset Locking Benefits: A key feature of this standard is enabling asset owners to use their NFTs, which represent locked assets, as collateral to secure loans. This approach allows asset owners to access liquidity without needing to withdraw their assets from pools or staking programs, thus preserving the associated benefits like interest accrual or voting rights.
+
+- Automated Loan and Collateral Management: The integration of automated features for managing the terms and conditions of the collateralized NFT is a deliberate choice to minimize transaction costs and complexity.
+
+- DeFi Composability: The strategic emphasis on DeFi composability, particularly the integration between asset-locking and collateralizing services, is pivotal for this standard. This approach aims to streamline the adoption of the standard across diverse DeFi platforms and services, fostering seamless connections within the DeFi ecosystem.
+
+### Alternate Designs and Related Work
+
+- Comparison with ERC-4907: While ERC-4907 also introduces a dual-role model for NFTs (owner and user), our standard focuses specifically on the use of NFTs for collateralization in financial transactions, diverging from ERC-4907’s rental-oriented approach.
+
+- Improvement Over Traditional Collateralization Methods: Compared to traditional DeFi collateralization, which often requires complete asset lock-up, this standard proposes a more dynamic and flexible model that allows for continued liquidity access.
+
+## Backwards Compatibility
+
+Fully compatible with ERC-721 and integrates with ERC-4907 for renting NFTs.
+
+## Test Cases
+
+### Test contract
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Code in the EIP itself _must_ be CC0-1.0 licensed. I permit a few other licenses (including `MIT`) in the assets directory, if you want to move this there.